src/components/sectionList/filters/useSectionListFilter.ts

```diff
@@ -0,0 +1,162 @@
+// this would translate to "token" in the old API, but does not exist in GIST-API
+export const IDENTIFIABLE_KEY = 'identifiable'
+
+const IDENTIFIABLE_FIELDS = {
+    name: {
+        operator: 'ilike',
+    },
+    code: {
+        operator: 'ilike',
+    },
+    shortName: {
+        operator: 'ilike',
+    },
+    id: {
+        operator: 'eq',
+    },
+}
+
+const getVerifiedFiltersForSchema = (
+    filters: ObjectParamType,
+    schema: Schema
+): Filters => {
+    if (!filters) {
+        return {}
+    }
+    /* TODO: verify values for filters */
+    const relevantFilters = Object.entries(filters).filter(([key]) => {
+        return key === IDENTIFIABLE_KEY || schema.properties[key]
+    })
+    return Object.fromEntries(relevantFilters)
+}
+
+const useFilterQueryParam = () => {
+    return useQueryParam('filter', CustomObjectParam)
+}
+
+export const useSectionListFilters = (): ReturnType<
+    typeof useQueryParam<Filters | null | undefined, Filters>
+> => {
+    const [filter, setFilter] = useFilterQueryParam()
+    const schema = useSchemaFromHandle()
+
+    return useMemo(
+        () => [getVerifiedFiltersForSchema(filter, schema), setFilter],
+        [filter, schema, setFilter]
+    )
+}
+
+export const useSectionListFilter = (
+    filterKey: string
+): [string | undefined, (value: string | undefined) => void] => {
+    const [filters, setFilters] = useSectionListFilters()
+
+    const boundSetFilter = useCallback(
+        (value: string | undefined) => {
+            if (!value) {
+                setFilters((filters) => {
+                    if (!filters) {
+                        return undefined
+                    }
+                    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+                    const { [filterKey]: _, ...rest } = filters
+                    return Object.keys(rest).length === 0 ? undefined : rest
+                })
+            } else {
+                setFilters((filters) => ({ ...filters, [filterKey]: value }))
+            }
+        },
+        [filterKey, setFilters]
+    )
+
+    return [filters?.[filterKey] ?? undefined, boundSetFilter]
+}
+
+export type ParseToQueryFilterResult = {
+    filter: string[]
+    rootJunction: GistParams['rootJunction']
+}
+
+const parseToQueryFilter = (filters: Filters): ParseToQueryFilterResult => {
+    const { [IDENTIFIABLE_KEY]: identifiableValue, ...restFilters } = filters
+    const queryFilters: string[] = []
+
+    const hasOtherFilters = Object.keys(restFilters).length > 0
+    // Groups are a powerful way to combine filters,
+    // here we use them for identifiable filters, to group them with "OR" and
+    // rest of the filters with "AND".
+    // Unfortunately, it doesn't work to use groups without at least two,
+    // so we need to add them conditionally.
+    // see https://docs.dhis2.org/en/develop/using-the-api/dhis-core-version-239/metadata-gist.html#gist_parameters_filter
```

Comment:
I've talked to Jan Bernitt about this, and he is working on a way for this to work even when sending a single group. That will make this code simpler, and then we don't have to deal with `rootJunction`.

Reply:
He already fixed that in master: https://github.com/dhis2/dhis2-core/pull/14501 , so we can remove most of the group logic here.